client/components/data/query-jetpack-plugins/index.jsx

```diff
@@ -15,7 +15,7 @@ class QueryJetpackPlugins extends Component {
 	};
 
 	componentDidMount() {
-		if ( this.props.siteIds && ! this.props.isRequestingForSites ) {
+		if ( this.props.siteIds.length && ! this.props.isRequestingForSites ) {
```

Comment:
Since `siteIds` is optional should use `this.props.siteIds?.length`?

Reply:
BTW, I am still getting two `https://public-api.wordpress.com/rest/v1.1/sites/[SiteId]/plugins` requests

Reply:
> Since siteIds is optional should we use this.props.siteIds?.length?

I was considering it as required given the propTypes:
```js
siteIds: PropTypes.arrayOf(
  PropTypes.oneOfType( [ PropTypes.string, PropTypes.number ] ).isRequired
).isRequired,
```

But figured out the previous version would not be a problem as the empty array would go to a `map` later down the line and **do nothing**.

> I am still getting two https://public-api.wordpress.com/rest/v1.1/sites/[SiteId]/plugins requests

I just found the actual problem.

1. When you don't have a selected site, this page is not firing any requests. After my change, it is firing only once. 
1. When you have a selected site, the page is already firing this request. After my change, it is firing twice.


I need to figure out where this first request for the selected site is being made, to assess how to best approach it.

Reply:
here the query is made only for the current site: https://github.com/Automattic/wp-calypso/blob/d3189b2d6fd95ec3c281f6f99be6112f60d5a256/client/my-sites/sidebar-unified/use-site-menu-items.js#L26-L33

Reply:
I think we can remove it from there and let each page / implementation decide if they want to fetch the plugins?

Reply:
I created another PR doing everything on the `use-site-menu-items`. What do you think?

PR: #60778

Reply:
I believe that `use-site-menu-items` shouldn't be fetching the plugins. For example, why users navigating to **Media** or **Home** would need the plugins? The addition of `dispatch( fetchPlugins( [ selectedSiteId ] ) ); ` to `use-site-menu-items` seems like a workaround that we need to avoid. 

Any page needing the site plugins should really use `<QueryJetpackPlugins siteIds={ siteIds } />` or something equivalent.

Reply:
Agreed, the fetch was removed from `use-menu-items` on https://github.com/Automattic/wp-calypso/pull/60707/commits/d071c4e7d7c0f2d32981409670e36c75c74f784f